remove remote/distributed classes

missedone opened the pull request:
moving the remote/distributed TestNG classes to separate project.
basically, this is prototype how we can remote distributed testng out.

see supplementary PR here: https://github.com/testng-team/testng-remote/pull/14

missedone committed:
remove remote/distributed classes

missedone commented on src/main/java/org/testng/TestNG.java:
```diff
@@ -1090,8 +1061,17 @@ else if (m_masterfileName == null) {
+   * @return
+   * @since 6.9.11 when moving distributed/remote classes out into separate project
+   */
+  protected List<ISuite> runSuites() {
```
introduce new override-able method `runSuites` to abstract how to run the suites, this is helpful for DistributedTestNG to override this method to run master/slave mode according to command line args, see also: https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/distributed/src/main/java/org/testng/distributed/DistributedTestNG.java#L54

missedone commented:
ping @cbeust , @juherr

cbeust commented:
That's a huge change. If you're telling me it's mostly files being moved around, I'll take your work and I'll merge it (pending @juherr's approval).

missedone commented:
yes, most of them just being moved.
the 'distributed testng' function was moved out of TestNG as well, it's available in testng-remote DistributedTestNG

juherr commented on src/main/java/org/testng/TestNG.java:
```diff
@@ -1577,8 +1555,6 @@ public void configure(Map cmdLineArgs) {
     result.xmlPathInJar = (String) cmdLineArgs.get(CommandLineArgs.XML_PATH_IN_JAR);
     result.junit = (Boolean) cmdLineArgs.get(CommandLineArgs.JUNIT);
     result.mixed = (Boolean) cmdLineArgs.get(CommandLineArgs.MIXED);
-    result.master = (String) cmdLineArgs.get(CommandLineArgs.MASTER);
```
Some warnings/exceptions are missing here to say that the master/slave feature moved into another jar.

juherr commented:
Everything seems good if we don't miss to transfert some files ;)

But merging this PR will have 1 big side effect: https://github.com/testng-team/testng-remote/issues/1 will have to be closed before the next TestNG release. Without it, we won't have workarround for the removed features.

juherr commented on src/main/java/org/testng/remote/RemoteArgs.java:
```diff
@@ -1,17 +0,0 @@
-
-import com.beust.jcommander.Parameter;
-
-public class RemoteArgs {
```
I don't see it into https://github.com/testng-team/testng-remote/tree/rm_distributed_classes/distributed/distributed/src/main/java/org/testng/remote

juherr commented on src/main/java/org/testng/xml/ResultContentHandler.java:
```diff
@@ -1,152 +0,0 @@
- *
- * @author Cedric Beust <cedric@beust.com>
- */
-public class ResultContentHandler extends DefaultHandler {
```
I don't find it on https://github.com/testng-team/testng-remote/tree/rm_distributed_classes

juherr commented on src/main/java/org/testng/xml/ResultXMLParser.java:
```diff
@@ -1,103 +0,0 @@
- *
- * @author Cedric Beust <cedric@beust.com>
- */
-public class ResultXMLParser extends XMLParser<Object> {
```
I don't find it on https://github.com/testng-team/testng-remote/tree/rm_distributed_classes

juherr commented:
Check: Except 2 classes, I found all others.

missedone commented:
thanks @juherr 

i just added the two missing classes here:
https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/remote/remote/src/main/java/org/testng/xml/ResultXMLParser.java
https://github.com/testng-team/testng-remote/blob/rm_distributed_classes/remote/remote/src/main/java/org/testng/xml/ResultContentHandler.java

note, for compatibility purpose, i modified ResultXMLParser.java to call parse method with reflection.

i'm wondering is ResultXMLParser still in used? i don't see any reference to it, @cbeust

cbeust commented:
Looks like we're ready to merge, @missedone can you do a rebase? If not I'll merge manually.

missedone commented:
hi @cbeust 
PR rebased
thanks
